apstools/devices.py

```diff
@@ -1620,6 +1620,117 @@ def calibrate_energy(self, value):
         self.use_set.put("Use")
 
 
+class PVPositionerSoftDone(PVPositioner):
+    """
+    PVPositioner that uses an internal "done" signal.
+
+    Parameters
+    ----------
+    prefix : str, optional
+        The device prefix used for all sub-positioners. This is optional as it
+        may be desirable to specify full PV names for PVPositioners.
+    readback_pv : str, optional
+        PV prefix of the readback signal. Disregarded if readback attribute is
+        created.
+    setpoint_pv : str, optional
+        PV prefix of the setpoint signal. Disregarded if setpoint attribute is
+        created.
+    tolerance : float, optional
+        Motion tolerance. The motion is considered done if
+        `abs(readback-setpoint) <= tolerance`. Defaults to `10^(-1*precision)`,
+        where `precision = setpoint.precision`
+    target_attr : str, optional
```

Comment:
Can this be refactored into an always-present attribute named `target` which is initially set to `None`?  Means more logic changes below but less likely to be misconfigured (or means more easily understood) by end-user of this device.

Reply:
in the `530-target` branch